packages/react-dom/src/test-utils/ReactTestUtilsAct.js

```diff
@@ -85,17 +85,19 @@ let actingUpdatesScopeDepth = 0;
 
 function act(callback: () => Thenable) {
   let previousActingUpdatesScopeDepth = actingUpdatesScopeDepth;
+  let previousActingUpdatesSigil;
   actingUpdatesScopeDepth++;
+  // we use the function flushPassiveEffects directly as the sigil,
+  // since it's unique to a renderer
   if (__DEV__) {
-    ReactShouldWarnActingUpdates.current = true;
+    previousActingUpdatesSigil = ReactActingRendererSigil.current;
+    ReactActingRendererSigil.current = flushPassiveEffects;
   }
 
   function onDone() {
     actingUpdatesScopeDepth--;
     if (__DEV__) {
-      if (actingUpdatesScopeDepth === 0) {
```

Comment:
I don't get how you were able to remove this depth check. Can you explain?

Reply:
ReactActingRendererSigil.current starts with a `.current` value of `null`. Inside `warnIfNotCurrentlyActingUpdatesInDev`, we check whether `.current === flushPassiveEffects)`
As the act() scope opens, it sets the value to flushPassiveEffects which isn't null, *until* the last scope, when previousActingUpdatesSigil gets set as null, triggering the warning.

We could have had the same pattern when the sigil was just a boolean, by saving `.current` as a `previousActingSigilBoolean` or something.

Reply:
Makes sense